From 7d213dde115b77ee5759ffffcdcd4a5026464d49 Mon Sep 17 00:00:00 2001
From: Frank Scheiner <frank.scheiner@web.de>
Date: Tue, 4 Aug 2026 15:48:08 +0200
Subject: Revert "ia64: Make acpi_cpufreq_cpu_exit return void"

From: Frank Scheiner <frank.scheiner@web.de>

Commit b4b1ddc9dfe9 ("cpufreq: Make cpufreq_driver->exit() return
void") was part of v6.6.145-rc1. But as it was derived from a
later Linux version w/o support for ia64 it didn't touch all
relevant parts in linux-6.6.y. Which now required an extra fix to
build correctly for ia64. But b4b1ddc9dfe9 was then not part of
the release v6.6.145. So the extra fix can go now, too.

This reverts commit bb51b626b5a8ef349b11876572540f19ad4d5030.

Signed-off-by: Frank Scheiner <frank.scheiner@web.de>
Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
---
 drivers/cpufreq/ia64-acpi-cpufreq.c |    4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

--- a/drivers/cpufreq/ia64-acpi-cpufreq.c
+++ b/drivers/cpufreq/ia64-acpi-cpufreq.c
@@ -302,7 +302,7 @@ acpi_cpufreq_cpu_init (
 }
 
 
-static void
+static int
 acpi_cpufreq_cpu_exit (
 	struct cpufreq_policy   *policy)
 {
@@ -316,6 +316,8 @@ acpi_cpufreq_cpu_exit (
 		kfree(policy->freq_table);
 		kfree(data);
 	}
+
+	return (0);
 }
 
 
